scripts/linters/pylint_extensions.py

```diff
@@ -1922,6 +1932,95 @@ def process_tokens(self, tokens):
                         'single-line-pragma', line=line_num)
 
 
+class TypeIgnoreCommentChecker(checkers.BaseChecker):
+    """Custom pylint checker which checks if MyPy's type ignores are properly
+    documented or not.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    EXCLUDED_DIRS_HAVING_IGNORE_TYPE_COMMENTS = copy.deepcopy(
+        EXCLUDED_TYPE_COMMENT_DIRECTORIES
+    )
+
+    name = 'type-ignore-comment'
+    priority = -1
+    msgs = {
+        'C0045': (
+            'Please try to Rule out the possibility of \'type: ignore\'.'
+            ' If \'type: ignore\' is really necessary, then add a proper'
+            ' comment.',
+            'mypy-ignore-used',
+            'MyPy ignores should be used with proper comments following'
+            ' the established patterns in the codebase. Except for'
+            ' no-untyped-call MyPy ignore.'
+        ),
+        'C0046': (
+            'Extra comment is present for MyPy type: ignore. Please'
+            ' remove it.',
+            'redundant-type-comment',
+            'No corresponding \'type: ignore\' is found for the type'
+            'ignore comment.'
+        )
+    }
+
+    def visit_module(self, node):
+        """Visit a module to ensure that there is a comment for each MyPy
+        type ignore.
+
+        Args:
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        module_excluded_from_comment_checks = False
+        for directory in self.EXCLUDED_DIRS_HAVING_IGNORE_TYPE_COMMENTS:
+            if re.search(directory, node.root().file):
+                module_excluded_from_comment_checks = True
+        if not module_excluded_from_comment_checks:
+            tokens = pylint_utils.tokenize_module(node)
+            self._process_module_tokens(tokens, node)
+
+    def _process_module_tokens(self, tokens, node):
+        """Custom pylint checker which allows only those MyPy type ignores
+        that are properly documented by a comment in the code. The comment
+        must be close to the actually ignore.
```

Comment:
```suggestion
        must be close to the actual ignore.
```

I think this is another example of why we better names to distinguish the documentation comments from the `type: ignore` comments

Reply:
Fixed -- wherever it is possible